### PR TITLE
proxy: fix goroutine leaks in tests

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

### CODE REVIEW

File: net/redisclient.go [hunk 1]
```diff
@@ -359,7 +359,9 @@ func (r *RedisRingClient) Close() {
 	r.once.Do(func() {
 		r.closed = true
 		close(r.quit)
-		r.ring.Close()
+		if r.ring != nil {
```
Comment (szuecs) [inline net/redisclient.go:362]: Ah re-introduced :)

File: proxy/fadeintesting_test.go [hunk 2]
```diff
@@ -107,18 +109,33 @@ func createDataClient(r ...*eskip.Route) fadeInDataClient {
...
+		return r, nil
+	case <-c.quit:
+		return nil, nil
+	}
 }
 
 func (c fadeInDataClient) LoadUpdate() ([]*eskip.Route, []string, error) {
-	return <-c.update, nil, nil
+	select {
+	// hm, blocking dataclient?
```
Comment (szuecs) [inline proxy/fadeintesting_test.go:129]: If you have 2 dataclients in use, what if only one fails such that 50% of your routes are not there. Do you want to startup normally or block and postpone/fail to start?
I think that's why we want to block in general.

**Comment by szuecs** [comment]
:+1:

File: net/redisclient.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply net/redisclient.go:362]: Yeah, I decided to plug nil check as quick fix but we should update ratelimit initialization such that it does not create redis client with nil ring

**Comment by AlexanderYastrebov** [comment]
:+1:

File: proxy/fadeintesting_test.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply proxy/fadeintesting_test.go:129]: Its a bit weird as we have polling cycle per dataclient but it works both ways

**Pull request merged**